components/eppp_link/eppp_eth.c

```diff
@@ -0,0 +1,249 @@
+    vTaskDelay(pdMS_TO_TICKS(STARTUP_DELAY_MS));
+    gpio_set_level(CONFIG_EPPP_LINK_ETHERNET_SINK_READY_GPIO, 1);
+#endif // CONFIG_EPPP_LINK_ETHERNET_CLK_SOURCE
+
+    // --- Initialize Ethernet driver ---
+    // Init common MAC and PHY configs to default
+    eth_mac_config_t mac_config = ETH_MAC_DEFAULT_CONFIG();
+    eth_phy_config_t phy_config = ETH_PHY_DEFAULT_CONFIG();
+
+    // Update PHY config based on board specific configuration
+    phy_config.reset_gpio_num = -1; // no HW reset
+
+    // Init vendor specific MAC config to default
+    eth_esp32_emac_config_t esp32_emac_config = ETH_ESP32_EMAC_DEFAULT_CONFIG();
+    // Update vendor specific MAC config based on board configuration
+    // No SMI, speed/duplex must be statically configured the same in both devices
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 3, 0)
+    esp32_emac_config.smi_gpio.mdc_num = -1;
+    esp32_emac_config.smi_gpio.mdio_num = -1;
+#else
+    esp32_emac_config.smi_mdc_gpio_num = -1;
+    esp32_emac_config.smi_mdio_gpio_num = -1;
+#endif
+#if CONFIG_EPPP_LINK_ETHERNET_CLK_SOURCE
+    esp32_emac_config.clock_config.rmii.clock_mode = EMAC_CLK_OUT;
+    esp32_emac_config.clock_config.rmii.clock_gpio = EMAC_CLK_OUT_180_GPIO;
+#else
+    esp32_emac_config.clock_config.rmii.clock_mode = EMAC_CLK_EXT_IN;
+    esp32_emac_config.clock_config.rmii.clock_gpio = EMAC_CLK_IN_GPIO;
+#endif // CONFIG_EPPP_LINK_ETHERNET_CLK_SOURCE
+
+    // Create new ESP32 Ethernet MAC instance
+    s_mac = esp_eth_mac_new_esp32(&esp32_emac_config, &mac_config);
+    // Create dummy PHY instance
+    s_phy = esp_eth_phy_new_dummy(&phy_config);
+
+    // Init Ethernet driver to default and install it
+    s_eth_handles = malloc(sizeof(esp_eth_handle_t));
+    ESP_RETURN_ON_FALSE(s_eth_handles, ESP_ERR_NO_MEM, TAG, "Our of memory");
+    esp_eth_config_t config = ETH_DEFAULT_CONFIG(s_mac, s_phy);
+#if CONFIG_EPPP_LINK_ETHERNET_CLK_SINK
+    // REF RMII CLK sink device performs multiple EMAC init attempts since RMII CLK source device may not be ready yet
+    int i;
+    for (i = 1; i <= 5; i++) {
+        ESP_LOGI(TAG, "Ethernet driver install attempt: %i", i);
+        if (esp_eth_driver_install(&config, &s_eth_handles[0]) == ESP_OK) {
+            break;
+        }
+        vTaskDelay(pdMS_TO_TICKS(100));
+    }
+    ESP_RETURN_ON_FALSE(i <= 5, ESP_FAIL, TAG, "Ethernet driver install failed");
+#else
+    ESP_RETURN_ON_ERROR(esp_eth_driver_install(&config, &s_eth_handles[0]), TAG, "Ethernet driver install failed");
+#endif // CONFIG_EPPP_LINK_ETHERNET_CLK_SINK
+    return ESP_OK;
+}
+
+void eppp_transport_deinit(void)
+{
+    if (esp_eth_driver_uninstall(&s_eth_handles[0]) != ESP_OK) {
+        ESP_LOGE(TAG, "Unable to deinitialize ethernet handle");
+    }
+    if (s_mac) {
+        s_mac->del(s_mac);
+        s_mac = NULL;
+    }
+    if (s_phy) {
+        s_phy->del(s_phy);
+        s_phy = NULL;
+    }
+    free(s_eth_handles);
+    s_eth_handles = NULL;
+}
+
+#else
```

Comment:
What about adding a comment that here we use `PHY chip` to make it clearer to users? And also add comment at the start of `#ifdef CONFIG_EPPP_LINK_ETHERNET_USE_DUMMY_PHY`. Or rename it to `EMAC to EMAC`?